File: 3819-count-covered-buildings/3819-count-covered-buildings.py
class Solution(object):
    def countCoveredBuildings(self, n, buildings):
        """
        :type n: int
        :type buildings: List[List[int]]
        :rtype: int
        """
        x_min, y_min = [float('inf')] * (n + 1), [float('inf')] * (n + 1)
        x_max, y_max = [float('-inf')] * (n + 1), [float('-inf')] * (n + 1)
        for building in buildings:
            x, y = building[0], building[1]
            x_min[x] = min(x_min[x], y)
            x_max[x] = max(x_max[x], y)
            y_min[y] = min(y_min[y], x)
            y_max[y] = max(y_max[y], x)

        return sum(1 if building[1] > x_min[building[0]] and building[1] < x_max[building[0]] and building[0] > y_min[building[1]] and building[0] < y_max[building[1]] else 0 for building in buildings)

File: 3819-count-covered-buildings/3819-count-covered-buildings.cs
public class Solution {
    public int CountCoveredBuildings(int n, int[][] buildings) {
        var x_min = new int[n + 1];
        var x_max = new int[n + 1];
        var y_min = new int[n + 1];
        var y_max = new int[n + 1];
        
        for (int i = 0; i < n + 1; i++) {
            x_min[i] = Int32.MaxValue;
            y_min[i] = Int32.MaxValue;
        }
        
        for (int i = 0; i < n + 1; i++) {
            x_max[i] = Int32.MinValue;
            y_max[i] = Int32.MinValue;
        }
        
        foreach (var building in buildings) {
            int x = building[0], y = building[1];
            x_min[x] = Math.Min(x_min[x], y);
            x_max[x] = Math.Max(x_max[x], y);
            y_min[y] = Math.Min(y_min[y], x);
            y_max[y] = Math.Max(y_max[y], x);
        }

        int cnt = 0;
        foreach (var building in buildings) {
            int x = building[0];
            int y = building[1];
            if (y > x_min[x] && y < x_max[x] && x > y_min[y] && x < y_max[y])
                cnt++;
        }
        return cnt;
    }
}